### PR TITLE
[ChopBox] [#101409884] Feature/multiple images upload

### PR DESCRIPTION
**Pull request by kola-er**
Hello,

Please review our work. We are done implementing the feature. We have also written the functional test which could do a single image upload for a chop at once due to Codeception 'attachFile()' function inability to attach more than one image. We tried using submitForm() too, but this doesn't attach file at all.
On the case of unit test, we didn't write that because we have no model to test on; all the logic for this feature are embedded in the controller. Moreover, we believe the functional test has taken care of that.

### CODE REVIEW

File: codeception.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ extensions:
 modules:
     config:
         Db:
-            dsn: 'mysql:host=127.0.0.1;dbname=chopbox_testing'
-            user: 'root'
-            password: ''
+            dsn: 'mysql:host=127.0.0.1;dbname=chopbox'
+            user: 'homestead'
+            password: 'secret'
```
Comment (unicodeveloper) [inline codeception.yml]: Why do you have all this here?. You already have this configuration in your _bootstrap.php . Please remove everything you recently added here

File: public/css/basic.css [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+.dropzone .dz-preview.dz-success .dz-success-mark {
+    display: block; }
+.dropzone .dz-preview .dz-error-mark, .dropzone .dz-preview .dz-success-mark {
+    position: absolute;
+    display: none;
+    left: 30px;
+    top: 30px;
+    width: 54px;
+    height: 58px;
+    left: 50%;
```
Comment (unicodeveloper) [inline public/css/basic.css]: Since you are no longer using dropzone, why do you have still have this here?

Comment (unicodeveloper) [inline public/css/basic.css]: I don't like the name given to this file. Why is it named `basic` ?

File: Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  #   push.app = "YOUR_ATLAS_USERNAME/YOUR_APPLICATION_NAME"
+  # end
+
+  # Enable provisioning with a shell script. Additional provisioners such as
+  # Puppet, Chef, Ansible, Salt, and Docker are also available. Please see the
+  # documentation for more information about their specific syntax and use.
+  # config.vm.provision "shell", inline: <<-SHELL
+  #   sudo apt-get update
+  #   sudo apt-get install -y apache2
+  # SHELL
```
Comment (unicodeveloper) [inline Vagrantfile]: Why are you committing your Vagrant File? Please remove

File: app/Http/Controllers/ChopsController.php [hunk 4]
```diff
@@ -16,107 +17,109 @@
...
 use Illuminate\Support\Facades\Auth;
 use Illuminate\Support\Facades\Redirect;
+use Illuminate\Support\Facades\Input as Input;
+use Validator;
+use Response;
 use Cloudder;
 
+\Cloudinary::config(array(
+    "cloud_name" => "chopbox"
+));
```
Comment (unicodeveloper) [inline app/Http/Controllers/ChopsController.php:30]: Why do you have this here?

File: app/Http/Controllers/ChopsController.php [hunk 5]
```diff
@@ -16,107 +17,109 @@
...
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  Request $request
+     *
+     * @return Response
+     */
+    public function store(ChopsFormRequest $request)
+    {
+        if (Input::file('image')) {
```
Comment (unicodeveloper) [inline app/Http/Controllers/ChopsController.php]: Assign Input::file('image') to a variable and just use the variable in other parts of your code

File: app/Http/Controllers/ChopsController.php [hunk 6]
```diff
@@ -16,107 +17,109 @@
...
+            $file = array();
+            $result = array();
+            $url = array();
+            $public_id = array();
```
Comment (unicodeveloper) [inline app/Http/Controllers/ChopsController.php]: Don't use this old style of declaring arrays, use [] instead

File: app/Http/Controllers/ChopsController.php [hunk 7]
```diff
@@ -16,107 +17,109 @@
...
+            $shortened_url = array();
+
+            // Declare an instance of Bitly
+            $this->shortener->setLogin(env('BITLY_LOGIN'));
+            $this->shortener->setKey(env('BITLY_API_KEY'));
+            $this->shortener->setFormat("json");
```
Comment (unicodeveloper) [inline app/Http/Controllers/ChopsController.php]: Abstract this to another function,...you can call that function `setConfig` or something like that and just call the function here

File: app/Http/Controllers/ChopsController.php [hunk 8]
```diff
@@ -16,107 +17,109 @@
...
+
+            // Upload each image to Cloudinary and shorten the url returned with Bitly.
+            for ($i = 0; $i < count(Input::file('image')); $i++) {
```
Comment (unicodeveloper) [inline app/Http/Controllers/ChopsController.php]: Performance will be slower, assign the size/count to a variable, and use the variable instead

File: app/Http/Controllers/ChopsController.php [hunk 9]
```diff
@@ -16,107 +17,109 @@
...
+            $upload->file_uri = $shortened_url[$i];
+            $upload->public_id = $public_id[$i];
+            $upload->chop_id = $this->chops->id;
+            $upload->user_id = $user->id;
+            $upload->save();
+        }
+
+		// Set a flash mesage to display on the page
+		$message = 'Success';
 		return redirect(route('chops.index', $message));
```
Comment (unicodeveloper) [inline app/Http/Controllers/ChopsController.php]: Not well indented, Fix!

File: app/Http/Controllers/ChopsController.php [hunk 10]
```diff
@@ -138,30 +141,24 @@ public function show($id)
...
-	public function update(Request $request, $id)
-	{
-		//
+        if($user->id === $owner_id) {
+            $chops = Chop::find($id);
+
+            return view('chops.updatechops', compact('chops', 'id'));
+        }
+        else
+            return redirect('chops');
```
Comment (unicodeveloper) [inline app/Http/Controllers/ChopsController.php]: Follow a convention and stick by it. Let the `else` have open and closing curly braces

File: app/Http/Controllers/ChopsController.php [hunk 11]
```diff
@@ -138,30 +141,24 @@ public function show($id)
 	 */
 	public function edit($id)
 	{
-		//
-	}
+        $owner_id = Chop::find($id)->user_id;
+        $user = Auth::user();
```
Comment (unicodeveloper) [inline app/Http/Controllers/ChopsController.php]: Assign `Chop::find($id)` to a variable and use that variable instead to avoid duplication.

**Comment by unicodeveloper** [comment]
Remove everything about dropzone since you no longer use it

File: public/css/basic.css (hunk 2 above)
Comment (kola-er) [inline public/css/basic.css]: Though I have removed it, that is how it is named straight from dropzone repository.

File: app/Http/Controllers/ChopsController.php (hunk 4 above)
Comment (kola-er) [inline app/Http/Controllers/ChopsController.php:30]: It supplies name of Cloudinary account for this app.
Even though this has been supplied in the env file, Cloudinary cl_image_tag() for retrieving and manipulating the images on view home.blade.php requests for it.

Comment (unicodeveloper) [inline app/Http/Controllers/ChopsController.php:30]: Pls remove it from there...that's dirty! :-1:  you can have a function that just retrieves it from the environment variable or from the Cloudinary config file

**Comment by kola-er** [comment]
Hello Prosper,

We have resolved all the issues you raised. Please have another check.

**Comment by unicodeveloper** [comment]
@andela-kerinoso Squash the last 4 commits into one commit. [Check here if you don't know how to squash](http://goodheads.io/2015/08/20/think-like-a-git-commit-like-a-boss-squash-like-a-pro/)

**Pull request closed without merging**